semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/ClasspathEntry.scala

```diff
@@ -0,0 +1,46 @@
+package scala.meta.internal.metacp
+
+import java.io.File
+import java.io.FilterInputStream
+import java.io.InputStream
+import java.nio.file.Files
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import scala.collection.mutable
+import scala.meta.io.AbsolutePath
+
+/** Represents a entry in a classpath that is either a package, file on disk or zip entry. */
```

Comment:
Given that, I suggest alternative names:
  * ClasspathEntry => ClasspathElement
  * ReadableClasspathEntry => Classfile
  * Package => Classdir (this class and classes below can become top-level, since their names become unique enough)
  * OnDisk => UncompressedClassfile
  * InJar => CompressedClassfile

Reply:
I like these suggestions. Will update the names 👍